base/random.jl

```diff
@@ -1166,6 +1166,30 @@ immutable UUID
 end
 UUID(u::AbstractString) = convert(UUID, u)
 
+function uuid1(rng::AbstractRNG=GLOBAL_RNG)
+    u = rand(rng, Uint128)
+
+    # mask off clock sequence and node
+    u &= 0x00000000000000003fffffffffffffff
+
+    # set the unicast/multicast bit
+    u |= 0x00000000000000000000010000000000
+
+    # 0x01b21dd213814000 is the number of milliseconds between
+    # the UUID epoch and Unix epoch
+    timestamp = round(Uint64, time() * 1e7) + 0x01b21dd213814000
```

Comment:
`time()` comes from [clock_now in support/timefuncs.c](https://github.com/JuliaLang/julia/blob/87a956bfb268172eee30ba737ef47cf839563c4d/src/support/timefuncs.c#L46-L56).  This has microsecond resolution on Unix, but currently only millisecond resolution on Windows.

Maybe you should patch src/support/timefuncs.c to use [ GetSystemTimePreciseAsFileTime](https://msdn.microsoft.com/en-us/library/windows/desktop/hh706895%28v=vs.85%29.aspx) on Windows?

(There is also some unnecessary rounding to/from floating-point; it might be nice to export an `iclock_now` function that returns an `Int64`.)